File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int y=0,a=x;
        if(x<0) return false;
        while(x>0){
            y=(x%10)+y*10;
            x/=10;
        }
        if(y==a){
            return true;
        }else{
            return false;
        }
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        int reverse = 0, actual = x;
        while (x > 0) {
            if (reverse > INT_MAX / 10 || (reverse == INT_MAX / 10 && x % 10 > 7))
                return false;
            reverse = reverse * 10 + (x % 10);
            x = x / 10;
        }
        return reverse==actual;
    }
};